galleon-plugins/src/main/java/org/wildfly/galleon/plugin/WfInstallPlugin.java

```diff
@@ -911,4 +912,61 @@ private static void changeLineEndings(final Path file, final boolean isWindows)
             Files.delete(temp);
         }
     }
+
+    public void appendFile(FileAppender config) throws ProvisioningException {
+        if(config.getTarget() == null) {
+            if(config.isIgnore()) {
+                return;
+            }
+            throw new ProvisioningException("Target can't be null when appending content");
+        }
+        final Path target = runtime.getStagedDir().resolve(config.getTarget());
+        if (!Files.exists(target)) {
+            if(config.isIgnore()) {
+                return;
+            }
+            throw new ProvisioningException(Errors.pathDoesNotExist(target));
+        }
+        try {
+            final Path src;
+            List<String> linesToAppend;
+            if (config.getSource() != null) {
+                src = runtime.getStagedDir().resolve(config.getSource());
+                if (!Files.exists(src)) {
+                    throw new ProvisioningException(Errors.pathDoesNotExist(src));
+                }
+                linesToAppend = Files.readAllLines(src);
+            } else {
+                src = null;
+                linesToAppend = config.getLines();
+            }
+            List<String> lines = Files.readAllLines(target);
+            List<String> updatedLines = new ArrayList<>(linesToAppend.size() + 3);
+            for (String line : lines) {
+                if (config.getMatch() != null && config.getMatch().matcher(line).find()) {
+                    for (int i = 0; i < linesToAppend.size(); i++) {
+                        if (i == 0) {
+                            updatedLines.add(line + linesToAppend.get(0));
+                        } else {
+                            updatedLines.add(linesToAppend.get(i));
+                        }
```

Comment:
Can you give a practical example where this kind of update is needed? Thanks.

Reply:
to append logger configurations to a logging.properties or to add lines to standalone.conf